### PR TITLE
Feature/71296 get area codes

### PR DESCRIPTION
**Pull request by hbjorgo**
Proxy for Area Codes.

### CODE REVIEW

File: src/Equinor.Procosys.Preservation.MainApi/AreaCode/MainApiAreaCodeService.cs [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+namespace Equinor.Procosys.Preservation.MainApi.AreaCode
+{
+    public class MainApiAreaCodeService : IAreaCodeApiService
+    {
+        private readonly string _apiVersion;
+        private readonly Uri _baseAddress;
+        private readonly IBearerTokenApiClient _mainApiClient;
+        private readonly IPlantApiService _plantApiService;
+        private readonly ILogger<MainApiAreaCodeService> _logger;
```
Comment (janid1967) [inline src/Equinor.Procosys.Preservation.MainApi/AreaCode/MainApiAreaCodeService.cs]: not in use. Log something or remove

File: src/Equinor.Procosys.Preservation.Query/GetAreaCodes/GetAreaCodesQueryHandler.cs [hunk 2]
```diff
@@ -0,0 +1,34 @@
+﻿using System;
```
Comment (janid1967) [inline src/Equinor.Procosys.Preservation.Query/GetAreaCodes/GetAreaCodesQueryHandler.cs]: remove 2 unused usings

File: src/tests/Equinor.Procosys.Preservation.MainApi.Tests/AreaCode/MainApiAreaCodeServiceTests.cs [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+                {
+                    Id = 3,
+                    Code = "CodeC",
+                    Description = "Description3",
+                }
+            };
+        }
+
+        [TestMethod]
+        public async Task GetAreaCodes_ReturnsThreeAreaCodes_TestAsync()
```
Comment (janid1967) [inline src/tests/Equinor.Procosys.Preservation.MainApi.Tests/AreaCode/MainApiAreaCodeServiceTests.cs]: remove _TestAsync() from names (all tests) ... adds no value to describe how test should behave

Comment (janid1967) [inline src/tests/Equinor.Procosys.Preservation.MainApi.Tests/AreaCode/MainApiAreaCodeServiceTests.cs]: add "Should" in from of second phrase (for all tests). GetAreaCodes_ShouldReturnsThreeAreaCodes

File: src/tests/Equinor.Procosys.Preservation.Query.Tests/GetAreaCodes/AreaCodeDtoTests.cs [hunk 4]
```diff
@@ -0,0 +1,18 @@
+﻿using Equinor.Procosys.Preservation.Query.GetAreaCodes;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace Equinor.Procosys.Preservation.Query.Tests.GetAreaCodes
+{
+    [TestClass]
+    public class AreaCodeDtoTests
+    {
+        [TestMethod]
+        public void Constructor_SetsProperties()
```
Comment (janid1967) [inline src/tests/Equinor.Procosys.Preservation.Query.Tests/GetAreaCodes/AreaCodeDtoTests.cs]: Constructor_ShouldSetProperties

File: src/tests/Equinor.Procosys.Preservation.Query.Tests/GetAreaCodes/GetAreaCodesQueryHandlerTests.cs [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+                },
+            };
+
+            _plantProviderMock = new Mock<IPlantProvider>();
+            _areaCodeApiServiceMock = new Mock<IAreaCodeApiService>();
+            _query = new GetAreaCodesQuery();
+        }
+
+        [TestMethod]
+        public async Task Handle_ReturnsCorrecResultType()
```
Comment (janid1967) [inline src/tests/Equinor.Procosys.Preservation.Query.Tests/GetAreaCodes/GetAreaCodesQueryHandlerTests.cs]: Handle_ShouldXXXX (all tests)

File: src/tests/Equinor.Procosys.Preservation.Query.Tests/GetAreaCodes/AreaCodeDtoTests.cs (hunk 4 above)
Comment (hbjorgo) [reply src/tests/Equinor.Procosys.Preservation.Query.Tests/GetAreaCodes/AreaCodeDtoTests.cs]: I'll rename to keep it consistent, but personally I'm not a big fan of Shouldxxx, as it indicates that it __should__ (bør) do something, but it's ok if it doesn't. When testing a spesific behavior, I expect that it actually does it.

**Pull request merged**